examples/md-flexible/src/Simulation.cpp

```diff
@@ -223,6 +223,8 @@ void Simulation::run() {
         const auto &boxMax = _autoPasContainer->getBoxMax();
         _autoPasContainer->addParticlesIf(emigrants, [&](auto &p) {
           if (autopas::utils::inBox(p.getR(), boxMin, boxMax)) {
+            // mark p as dummy in the emigrants vector, so it will be kicked out via remove_if later.
+            // It's still added to autopas here and its ownership state set by addParticle().
```

Comment:
No, not really. Here is what happens:
1. We get the emigrants from `updateContainer`.
2. We update the size of the local domain according to the domain decomposition load balancing update. `resizeBox` gives us the particles that are now also outside of the domain.
After this resize, some emigrants from 1. might be inside the new box boundaries.
3. We use `addParticlesIf` to add these emigrants into AutoPas.
Those that are added now have to be removed from the `emigrants` vector because they are processed. We don't do this one by one because this is highly inefficient (it would require reshuffling the vector after every delete). Hence, we mark those that are added as dummies in the vector. The container (soon the `LogicHandler`) ensures that the particle that is actually added will have its OwnershipState correctly set to `owned` since we always add copies.
4. We use `remove_if` to efficiently clear all dummies from the `emigrants` vector.
5. We merge the remaining emigrants with the particles that dropped out of the box due to the resize.

So how do you suggest we explain this better in the code without creating a huge wall of text? :)

Reply:
But now that I look at it, this is missing an `erase()`

Reply:
Ah shit, I somehow completely overlooked the `return false` statement...

I would still change the comment to be more straightforward. Maybe like this?
```suggestion
            // Reinsert emigrants that might still be in the domain after load balancing.
          // Reinserted particles are marked as dummy here, so they can be erased from emigrants below.
```

But staying an annoyance here. You mentioned before that an inserted particle's ownership state is set by `addParticle()`. I can't find where this is?

Reply:
Your first new sentence is basically the comment that is already at the start of this code block. Maybe we should look at it as a whole?

The enforcement of ownership is currently not very consistent. Most containers enforce halo ownership in their `addHaloParticleImpl(),` not sure if any enforce `owned` in any `addParticleImpl()`, but anything `FullParticleCell`-based will check that the inserted particle has compatible ownership. Manish's PR https://github.com/AutoPas/AutoPas/pull/821 will move this to the `LogicHandler` for both versions and finally bring sanity to this.